eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -158,40 +185,45 @@ stages:
                   -ArtifactsList ('$(ArtifactsJson)' | ConvertFrom-Json | Select-Object name, groupId, uberJar)
 
-        - deployment: TagRepository
+        - job: TagRepository
           displayName: "Create release tag"
           condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
-          environment: ${{ parameters.Environment }}
           dependsOn: VerifyReleaseVersion
           pool:
             name: azsdk-pool-mms-win-2022-general
             image: azsdk-pool-mms-win-2022-1espt
             os: windows
-          strategy:
-            runOnce:
-              deploy:
-                steps:
-                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
-                  - template: /eng/common/pipelines/templates/steps/retain-run.yml
-                  # No pattern, just download everything that's been signed
-                  - download: current
-                    displayName: 'Download Artifact: ${{parameters.ArtifactName}}-signed'
-                    artifact: ${{parameters.ArtifactName}}-signed
-                  - ${{ each artifact in parameters.Artifacts }}:
-                    - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
-                      parameters:
-                        ArtifactLocation: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
-                        PackageRepository: Maven
-                        ReleaseSha: $(Build.SourceVersion)
-
-        - deployment: PublishPackage
-          displayName: "Publish to Maven Central"
+          steps:
+            - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+            - template: /eng/common/pipelines/templates/steps/retain-run.yml
+            # No pattern, just download everything that's been signed
+            - download: current
+              displayName: 'Download Artifact: ${{parameters.ArtifactName}}-signed'
+              artifact: ${{parameters.ArtifactName}}-signed
+            - ${{ each artifact in parameters.Artifacts }}:
+              - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
+                parameters:
+                  ArtifactLocation: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
+                  PackageRepository: Maven
+                  ReleaseSha: $(Build.SourceVersion)
+
+        - deployment: PublishESRPPackage
+          displayName: "Publish to Maven Central via ESRP"
           condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'))
           # This timeout shouldn't be necessary once we're able to parallelize better. Right now,
           # this is here to ensure larger areas (30+) libraries don't time out.
           timeoutInMinutes: 120
           environment: ${{ parameters.Environment }}
```

Comment:
potential clean-up: We shouldn't need to have variable environments we should hard-code to `package-publish` in all the places we do this deployment.